posthog/api/capture.py

```diff
@@ -551,6 +569,89 @@ def get_event(request):
     return cors_response(request, JsonResponse({"status": 1}))
 
 
+def replace_with_warning(event: dict[str, Any]) -> dict[str, Any] | None:
+    """
+    Replace the event with a warning message if the event is too large to be sent to Kafka.
+    The event passed in should be safe to discard (because we know kafka won't accept it).
+    We do this so that when we're playing back the recording we can insert useful info in the UI.
+    """
+    try:
+        #
+        properties = event.pop("properties", {})
+        snapshot_items = properties.pop("$snapshot_items", [])
+        # since we had message too large there really should be an item in the list
+        # but just in case, since we would have dropped this anyway
+        if not snapshot_items:
+            return None
+
+        first_item = snapshot_items[0]
+        if not isinstance(first_item, dict) or ("$window_id" not in first_item and "timestamp" not in first_item):
+            return None
+
+        return {
+            **event,
+            "properties": {
+                **properties,
+                "$snapshot_bytes": 0,
+                "$snapshot_items": [
+                    {
+                        "type": 5,
+                        "data": {
+                            "tag": "Message too large",
+                        },
+                        "$window_id": first_item.get("$window_id"),
+                        "timestamp": first_item.get("timestamp"),
+                    }
+                ],
+            },
+        }
+    except Exception as ex:
+        capture_exception(ex)
```

Comment:
I'd love that all events captured during replay ingestion would have a tag such as:
scope.set_tag("capture-pathway", "replay")
so we could easily query/have a dashboard for all errors happening in this part.